README.md

```diff
@@ -38,33 +40,82 @@ On your `_ViewImports.cshtml` add the `using` and TagHelper entries:
 
 ```c#
-private Canvas2dContext _context;
+private Canvas2DContext _context;
 
 protected BECanvasComponent _canvasReference;
 
-protected override void OnAfterRender()
+protected override async Task OnAfterRenderAsync()
 {
-    this._context = this._canvasReference.CreateCanvas2d();
-    this._context.FillStyle = "green";
+    this._context = await this._canvasReference.CreateCanvas2DAsync();
+    await this._context.SetFillStyleAsync("green");
+
+    await this._context.FillRectAsync(10, 100, 100, 100);
+
+    await this._context.SetFontAsync("48px serif");
+    await this._context.StrokeTextAsync("Hello Blazor!!!", 10, 100);
+}
+```
+
+**NOTE** You cannot call `CreateCanvas2DAsync` in `OnInitAsync`, because the underlying `&lt;canvas&gt;` element is not yet present in the generated markup.
+
+### WebGL
 
-    this._context.FillRect(10, 100, 100, 100);
+On your component C# code (regardless if inline on .razor or in a .cs file), from a `BECanvasComponent` reference, create a `WebGLContext`, and then use the context methods to draw on the canvas:
+
+```c#
+private WebGLContext _context;
+
+protected BECanvasComponent _canvasReference;
 
-    this._context.Font = "48px serif";
-    this._context.StrokeText("Hello Blazor!!!", 10, 100);
+protected override async Task OnAfterRenderAsync()
+{
+    this._context = await this._canvasReference.CreateWebGLAsync(new WebGLContextAttributes
+    {
+        PowerPreference = WebGLContextAttributes.POWER_PREFERENCE_HIGH_PERFORMANCE
+    });
+    
+    await this._context.ClearColorAsync(0, 0, 0, 1);
+    await this._context.ClearAsync(BufferBits.COLOR_BUFFER_BIT);
 }
 ```
 
+**NOTE** You cannot call `CreateWebGLAsync` in `OnInitAsync`, because the underlying `&lt;canvas&gt;` element is not yet present in the generated markup.
+
+### Call Batching
+
+All javascript interop are batched as needed to improve performance. In high-performance scenarios this behavior will not have any effect: each call will execute immediately. In low-performance scenarios, consective calls to canvas APIs will be queued. JavaScript interop calls will be made with each batch of queued commands sequentially, to avoid the performance impact of multiple concurrent interop calls.
+
+When using server-side Razor Components, because of the server-side rendering mechanism, only the last drawing operation executed will appear to render on the client, overwriting all previous operations. In the example code above, for example, drawing the triangles would appear to "erase" the black background drawn immediately before, leaving the canvas transparent.
+
+To avoid this issue, all WebGL **drawing** operations should be explicitly preceded and followed by `BeginBatchAsync` and `EndBatchAsync` calls.
+
+For example:
+
+```c#
+await this._context.ClearColorAsync(0, 0, 0, 1); // this call does not draw anything, so it does not need to be included in the explicit batch
+
+await this._context.BeginBatchAsync(); // begin the explicit batch
```

Comment:
I'm not sure how easy it would be but what if you make this return an ``IDisposable`` which calls ``EndBatchAsync`` on disposal. The syntax would look something like this:

```cs
using (this._context.BeginBatchAsync()) {
  await this._context.ClearAsync(BufferBits.COLOR_BUFFER_BIT);
  await this._context.DrawArraysAsync(Primitive.TRIANGLES, 0, 3);
}
```

However I'm not sure if this is going to work because of all the ``async`` stuff